MatrixIO.IO/IO/ConstrainedStream.cs

```diff
@@ -5,155 +5,145 @@
+            }
 
-            if (offset < _CurrentConstraint.Start)
+            if (offset + length > _currentConstraint.End)
             {
-                Trace.WriteLine(string.Format("Attempted to set a new constraint ({0},{1}) offset outside the bounds of the existing constraint({2},{3}).", offset, length, _CurrentConstraint.Start, _CurrentConstraint.Length), "WARNING");
-                offset = _CurrentConstraint.Start;
+                Trace.WriteLine($"Attempted to set a new constraint ({offset},{length}) length outside the bounds of the existing constraint({_currentConstraint.Start},{_currentConstraint.Length}).", "WARNING");
+                length = _currentConstraint.End - offset;
             }
-            if (offset + length > _CurrentConstraint.End)
+
+            if (_currentConstraint != null)
             {
-                Trace.WriteLine(string.Format("Attempted to set a new constraint ({0},{1}) length outside the bounds of the existing constraint({2},{3}).", offset, length, _CurrentConstraint.Start, _CurrentConstraint.Length), "WARNING");
-                length = _CurrentConstraint.End - offset;
+                _ConstraintStack.Push(_currentConstraint);
             }
 
-            if (_CurrentConstraint != null) _ConstraintStack.Push(_CurrentConstraint);
-            _CurrentConstraint = new ByteRange(offset, offset + length);
+            _currentConstraint = new ByteRange(offset, offset + length);
         }
 
         public void PopConstraint()
         {
-            if (_ConstraintStack.Count > 0) _CurrentConstraint = _ConstraintStack.Pop();
+            if (_ConstraintStack.Count > 0) _currentConstraint = _ConstraintStack.Pop();
             else
             {
                 Trace.WriteLine("Attempted to pop constraint too many times.", "WARNING");
-                _CurrentConstraint = ByteRange.MaxValue;
+                _currentConstraint = ByteRange.MaxValue;
             }
         }
 
-        public override bool CanRead
-        {
-            get { return _BaseStream.CanRead; }
-        }
+        public override bool CanRead => _baseStream.CanRead;
 
-        public override bool CanSeek
-        {
-            get { return _BaseStream.CanSeek; }
-        }
+        public override bool CanSeek => _baseStream.CanSeek;
 
-        public override bool CanWrite
-        {
-            get { return _BaseStream.CanWrite; }
-        }
+        public override bool CanWrite => _baseStream.CanWrite;
 
         public override void Flush()
         {
-            _BaseStream.Flush();
+            _baseStream.Flush();
         }
 
-        public override long Length
-        {
-            get 
-            { 
-                return Math.Min(_BaseStream.Length, _CurrentConstraint.End); 
-            }
-        }
+        public override long Length => Math.Min(_baseStream.Length, _currentConstraint.End);
 
-        public long _Count;
+        public long _count;
```

Comment:
This probably shouldn't be publicly exposed like this.

Reply:
Good catch. #resolved